client/src/test/java/com/sinch/sdk/domains/voice/adapters/CallsServiceTest.java

```diff
@@ -30,67 +26,58 @@
 @TestWithResources
 public class CallsServiceTest extends BaseTest {
 
-  @Mock CallsApi api;
-  @Mock VoiceContext context;
-  @Mock HttpClient httpClient;
-  @Mock Map<String, AuthManager> authManagers;
+  @Mock com.sinch.sdk.domains.voice.api.v1.CallsService v1;
 
   @Captor ArgumentCaptor<String> callIdCaptor;
-  @Captor ArgumentCaptor<String> callLegCaptor;
-  @Captor ArgumentCaptor<SVAMLRequestBodyDto> updateParametersCaptor;
+  @Captor ArgumentCaptor<CallLeg> callLegCaptor;
+  @Captor ArgumentCaptor<SvamlControl> updateParametersCaptor;
   CallsService service;
 
   @BeforeEach
   public void initMocks() {
-    service = spy(new CallsService(context, httpClient, authManagers));
-    doReturn(api).when(service).getApi();
+    service = spy(new CallsService(v1));
   }
 
   @Test
   void get() throws ApiException {
-
-    when(api.callingGetCallResult(
-            eq(CallsResponseDtoTest.expectedCallsGetInformationResponseDto.getCallId())))
-        .thenReturn(CallsResponseDtoTest.expectedCallsGetInformationResponseDto);
+    when(v1.get(eq(CallInformationTest.expectedCallsGetInformationResponseDto.getCallId())))
+        .thenReturn(CallInformationTest.expectedCallsGetInformationResponseDto);
 
     CallInformation response =
-        service.get(CallsResponseDtoTest.expectedCallsGetInformationResponseDto.getCallId());
+        service.get(CallInformationTest.expectedCallsGetInformationResponseDto.getCallId());
 
-    Assertions.assertThat(response)
-        .usingRecursiveComparison()
-        .isEqualTo(CallsDtoConverterTest.expectedCallInformation);
+    TestHelpers.recursiveEquals(response, CallsDtoConverterTest.expectedCallInformation);
   }
 
   @Test
   void update() throws ApiException {
-
     service.update("call id", SVAMLControlTest.parameters);
 
-    verify(api).callingUpdateCall(callIdCaptor.capture(), updateParametersCaptor.capture());
+    verify(v1).update(callIdCaptor.capture(), updateParametersCaptor.capture());
 
     String callId = callIdCaptor.getValue();
     Assertions.assertThat(callId).isEqualTo("call id");
 
-    SVAMLRequestBodyDto body = updateParametersCaptor.getValue();
-    Assertions.assertThat(body).isEqualTo(CallsDtoConverterTest.svamlRequestBodyDto);
+    SvamlControl body = updateParametersCaptor.getValue();
+    TestHelpers.recursiveEquals(body, CallsDtoConverterTest.svamlControlDto);
   }
 
   @Test
   void manageWithCallLeg() throws ApiException {
 
     service.manageWithCallLeg("call id", CallLegType.BOTH, SVAMLControlTest.parameters);
```

Comment:
Should it be `CallLegType` or `CallLeg` (from v1) ?

Reply:
Not here: we are testing "old" design service still working: https://developers.sinch.com/java-sdk/apidocs/com/sinch/sdk/domains/voice/CallsService.html#manageWithCallLeg(java.lang.String,com.sinch.sdk.domains.voice.models.CallLegType,com.sinch.sdk.domains.voice.models.svaml.SVAMLControl)
So it accept a `CallLegType` and should be wrapped internally onto "new design" service based onto CallLeg.
This is why :
- https://github.com/sinch/sinch-sdk-java/pull/164/files#diff-be39bcdae0377a57e69f1a79400a0628c221f8c3859acfd3fd970eb23c0e7ef7R32
- and https://github.com/sinch/sinch-sdk-java/pull/164/files#diff-be39bcdae0377a57e69f1a79400a0628c221f8c3859acfd3fd970eb23c0e7ef7R77

We are validating old design CallLegtype was correctly transformed to new `CallLeg` (type and value)

Reply:
Thanks for your explanation!